src/execution/mod.rs

```diff
@@ -208,17 +211,33 @@ impl<'b> RuntimeInstance<'b> {
                     stack.push_value(res.into());
                 }
                 // i32.mul: [i32 i32] -> [i32]
-                0x6C => {
-                  let v1: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
-                  let v2: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
-                  let res = v1.wrapping_mul(v2);
+                Op1Byte::I32Mul => {
+                    let v1: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
+                    let v2: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
+                    let res = v1.wrapping_mul(v2);
+
+                    trace!("Instruction: i32.mul [{v1} {v2}] -> [{res}]");
+                    stack.push_value(res.into());
+                    let res = v1.wrapping_mul(v2);
 
-                  trace!("Instruction: i32.mul [{v1} {v2}] -> [{res}]");
-                  stack.push_value(res.into());
+                    trace!("Instruction: i32.mul [{v1} {v2}] -> [{res}]");
+                    stack.push_value(res.into());
                 }
-                other => {
-                    trace!("Unknown instruction {other:#x}, skipping..");
+                Op1Byte::FCInstructions => {
+                    let second_byte = wasm.read_u8().unwrap_validated();
+                    let parsed_2byte_instr =
+                        Op2Byte::new(u16::from_be_bytes([first_byte, second_byte]));
+                    match parsed_2byte_instr {
```

Comment:
I'm not a fan of the nested match statements

Reply:
Okay, then how do you say we handle the fact that some instructions are 2 bytes? In that case we have to always read a u16. If there's another better way of doing this, I'm all ears.